src/core/sock.c

```diff
@@ -313,6 +314,11 @@ static int nn_sock_setopt_inner (struct nn_sock *self, int level,
                 return -EINVAL;
             dst = &self->rcvbuf;
             break;
+        case NN_RCVMAXSIZE:
+            if (nn_slow (val <= -1))
```

Comment:
Well, I prefer to use specific -1 as the documented case, in case we ever want to have anything else in the future (unlikely, I know).  It will also catch incorrect attempts to set some value that is too large (via rollover).